syncserver/src/web/handlers.rs

```diff
@@ -30,24 +30,42 @@ use crate::{
     },
 };
 
+use glean::server_events::{EventsPing, RequestInfo, SyncstorageGetCollectionsEvent};
+
 pub const ONE_KB: f64 = 1024.0;
 
 pub async fn get_collections(
     meta: MetaRequest,
     db_pool: DbTransactionPool,
     request: HttpRequest,
+    state: Data<ServerState>,
 ) -> Result<HttpResponse, ApiError> {
-    // The values below, prefixed by `_`, are temporarily and intentionally ignored at present.
-    // They will be passed to the Glean logic we will implement to emit metrics.
-    // We'd like for the data to be ready and in place to pass to that logic.
-    let _hashed_fxa_uid: String = meta.user_id.hashed_fxa_uid.clone();
-    let _hashed_device_id: String = meta.user_id.hashed_device_id.clone();
-    let user_agent = request
-        .headers()
-        .get(header::USER_AGENT)
-        .and_then(|header| header.to_str().ok())
-        .unwrap_or("none");
-    let _device_info = get_device_info(user_agent);
+    if state.glean_enabled {
+        // Values below are be passed to the Glean logic to emit metrics.
+        // This is used to measure DAU (Daily Active Use) of Sync.
+        let hashed_fxa_uid: String = meta.user_id.hashed_fxa_uid.clone();
+        let hashed_device_id: String = meta.user_id.hashed_device_id.clone();
+        let user_agent = request
+            .headers()
+            .get(header::USER_AGENT)
+            .and_then(|header| header.to_str().ok())
+            .unwrap_or("none");
+        let device_info: DeviceInfo = get_device_info(user_agent);
+
+        state.glean_logger.record_events_ping(
+            &RequestInfo {
+                user_agent: user_agent.to_owned(),
+                ip_address: "".to_owned(),
+            },
+            &EventsPing {
+                syncstorage_device_family: device_info.device_family.to_string(),
+                syncstorage_hashed_device_id: hashed_device_id.to_string(),
+                syncstorage_hashed_fxa_uid: hashed_fxa_uid.to_string(),
```

Comment:
you already have ownership of these Strings here so I don't think they need to_string'ing (or probably even the cloning as well?)
```suggestion
        let user_agent = request
            .headers()
            .get(header::USER_AGENT)
            .and_then(|header| header.to_str().ok())
            .unwrap_or("none");
        let device_info: DeviceInfo = get_device_info(user_agent);

        state.glean_logger.record_events_ping(
            &RequestInfo {
                user_agent: user_agent.to_owned(),
                ip_address: "".to_owned(),
            },
            &EventsPing {
                syncstorage_device_family: device_info.device_family.to_string(),
                syncstorage_hashed_device_id: meta.user_id.hashed_device_id,
                syncstorage_hashed_fxa_uid: meta.user_id.hashed_fxa_uid,
```

Reply:
Also I wonder if an empty string for the lack of user agent would be preferable to "none"